src/screens/NuxLocationScreen.js

```diff
@@ -0,0 +1,192 @@
+import Fragment from '../common/Fragment';
+import LoadingIndicator from '../common/LoadingIndicator';
+import LoaderComponent from '../common/LoaderComponent';
+import Header from '../common/Header';
+import Container from '../common/Container';
+import LocationPicker from '../components/LocationPicker';
+import { COLORS, TYPOGRAPHY } from '../theme';
+
+const styles = StyleSheet.create({
+  container: {
+    backgroundColor: COLORS.primary,
+    flex: 1,
+    justifyContent: 'center',
+    paddingVertical: 30,
+  },
+  descriptionText: {
+    ...TYPOGRAPHY.heading,
+    color: COLORS.textInverse,
+    paddingHorizontal: 15,
+    paddingVertical: 30,
+    textAlign: 'center',
+  },
+  iconContainer: {
+    alignSelf: 'center',
+  },
+  input: {
+    color: COLORS.textInverse,
+  },
+  label: {
+    color: COLORS.textInverse,
+  },
+  validationText: {
+    color: COLORS.danger2,
+  },
+});
+
+type InjectedProps = {|
+  onContinuePress: (location?: Location) => void | Promise<any>,
+|};
+
+@flatNavigationParamsAndScreenProps
+@observer
+class NuxLocation extends InjectedComponent<InjectedProps> {
+  // We use LocationStore.count() in boundaries  of this component in two places
+  // here and in LocationPicker. LocationPicker flushes the cache of LocationStore
+  // and it leads to infinite loop, so we freeze countLoader for the current case.
+  // To avoid this crutch we need to implement more robust cache control in our
+  // DAO
+  @observable _freezedLocationsCountLoader: LoadObject<number> = null;
+
+  componentDidMount() {
+    when(
+      () => this._locationsCountLoader.hasValue(),
+      () => {
+        this._freezedLocationsCountLoader = this._locationsCountLoader;
+      },
+    );
+  }
+
+  @computed
+  get _locationsCountLoader(): LoadObject<number> {
+    return this._freezedLocationsCountLoader || LocationStore.count();
+  }
+
+  render() {
+    return (
+      <LoaderComponent
+        loadedComponent={LoadedComponent}
+        loader={this._locationsCountLoader}
+        loadingComponent={LoadingComponent}
+        onContinuePress={this.injectedProps.onContinuePress}
+      />
+    );
+  }
+}
+
+const LoadingComponent = () => (
+  <Fragment>
+    <Header showBackButton title="1. Setup location" />
+    <LoadingIndicator color={COLORS.secondary} style={styles.container} />
+  </Fragment>
+);
+
+type LoadedComponentProps = {|
+  onContinuePress: (location?: Location) => void | Promise<any>,
+  value: number,
+|};
+
+@observer
+class LoadedComponent extends React.Component<LoadedComponentProps> {
+  _isLocationLoadingToggleStore: ToggleStore = new ToggleStore();
+
+  @observable _selectedLocation: ?Location = null;
+
+  @action
+  _onLocationSelect = (location: ?Location) => {
+    this._selectedLocation = location;
+  };
+
+  _onContinuePress = async (): Promise<void> => {
+    const { value: locationsCount, onContinuePress } = this.props;
+    switch (locationsCount) {
```

Comment:
I think an `if` statement would be better than a switch for readability. It's not usually a good idea to have case statements as hard-coded values. They are usually constants so it's easier to understand what each case means.